maltoolbox/attackgraph/attackgraph.py

```diff
@@ -576,42 +599,67 @@ def _generate_graph(self) -> None:
-                               '"%s" to link with for attack step "%s"(%d)!')
-                        logger.error(
-                            msg,
-                            target_node_full_name,
-                            ag_node.full_name,
-                            ag_node.id
-                        )
-                        raise AttackGraphStepExpressionError(
-                            msg % (
-                                target_node_full_name,
-                                ag_node.full_name,
-                                ag_node.id
-                            )
+
+            if not ag_node.asset:
+                raise AttackGraphException('Attack graph node is missing '
+                    'asset link')
+            lang_graph_asset = self.lang_graph.assets[ag_node.asset.type]
+
+            lang_graph_attack_step = lang_graph_asset.attack_steps[\
+                ag_node.name]
+
+            while lang_graph_attack_step:
+                for child in lang_graph_attack_step.children:
+                    for target_attack_step, expr_chain in \
+                            lang_graph_attack_step.children[child]:
+                        target_assets = self._follow_expr_chain(
+                            self.model,
+                            [ag_node.asset],
+                            expr_chain
                         )
-                    ag_node.children.append(target_node)
-                    target_node.parents.append(ag_node)
+
+                        for target_asset in target_assets:
+                            if target_asset is not None:
+                                target_node_full_name = target_asset.name + \
+                                    ':' + target_attack_step.name
+                                target_node = self.get_node_by_full_name(
+                                    target_node_full_name)
+                                if target_node is None:
+                                    msg = ('Failed to find target node '
+                                           '"%s" to link with for attack '
+                                           'step "%s"(%d)!')
+                                    logger.error(
+                                        msg,
+                                        target_node_full_name,
+                                        ag_node.full_name,
+                                        ag_node.id
+                                    )
+                                    raise AttackGraphStepExpressionError(
+                                        msg % (
+                                            target_node_full_name,
+                                            ag_node.full_name,
+                                            ag_node.id
+                                        )
+                                    )
+
+                                assert ag_node.id is not None
```

Comment:
This seems a bit out of place here. `ag_node` is not altered in this loop at all, right?